### PR TITLE
Enable YANG data range and length validation

### PR DESCRIPTION
**Pull request by wingo**

### CODE REVIEW

**Comment by wingo** [comment]
PTAL @xray7224 / @dpino .  Thanks :)  (Fixes #697.)

**Review by dpino — APPROVED** [review]

File: src/lib/yang/schema.lua [hunk 1]
```diff
@@ -47,20 +47,28 @@ local function require_argument(loc, argument)
...
+      end
+      if r ~= 'max' and r < 0 and kind == 'length' then
+         error("length argument may not be negative: "..r)
+      end
+      if l ~= 'min' and r ~= 'max' and r < l then
+         error("invalid "..kind..": "..part)
+      end
       return { l, r }
    end
    local parts = range:split("|")
```
Comment (dpino) [inline src/lib/yang/schema.lua]: This line can be removed as it's unused.

**Comment by dpino** [comment]
`lib.yang.schema` fails cause now a range is an array of ranges. I'm not sure whether that's the intended behaviour.

```
diff --git a/src/lib/yang/schema.lua b/src/lib/yang/schema.lua
index 3da127b..940f40e 100644
--- a/src/lib/yang/schema.lua
+++ b/src/lib/yang/schema.lua
@@ -1069,8 +1069,8 @@ function selftest()
    assert(contents.body["name"].description == "Name of fruit.")
    assert(contents.body["score"].type.id == "uint8")
    assert(contents.body["score"].mandatory == true)
-   assert(contents.body["score"].type.range.value[1] == 0)
-   assert(contents.body["score"].type.range.value[2] == 10)
+   assert(contents.body["score"].type.range.value[1][1] == 0)
+   assert(contents.body["score"].type.range.value[1][2] == 10)
```

**Comment by wingo** [comment]
This test error looks spurious :-(

**Review by dpino — APPROVED** [review]

**Pull request merged**